src/MeshMaterialSplitter.cpp

```diff
@@ -62,12 +63,18 @@ bool operator==(const ContourKey& key1, const ContourKey& key2)
  * @param mesh The mesh to fill the voxels grid with
  * @param texture_data_provider The provider containing the painted texture data
  * @param voxel_grid The voxels grid to be filled with mesh data
+ * @param main_extruder The main extruder of the mesh, to be used when we find a texture part that uses a disabled/inexisting extruder
  * @return True if this generated relevant data for multi-extruder, otherwise this means the mesh is completely filled with only extruder 0 and there is no need to go further on
  *         trying to calculate the modified meshes.
  */
-bool makeVoxelGridFromTexture(const Mesh& mesh, const std::shared_ptr<TextureDataProvider>& texture_data_provider, VoxelGrid& voxel_grid)
+bool makeVoxelGridFromTexture(const Mesh& mesh, const std::shared_ptr<TextureDataProvider>& texture_data_provider, VoxelGrid& voxel_grid, const size_t main_extruder)
 {
     boost::concurrent_flat_set<uint8_t> found_extruders;
+    std::vector<size_t> active_extruders;
```

Comment:
Very minor point, but since we're only using`active_extruders` to check if an extruder is present a `std::vector` isn't optimal. Instead we can use an `unordered_set` or an fixed length array of `[false; 16]` where only the active extruders are set to `true`.